File: Easy/MIME Type/C#/MIME Type.cs
﻿using System;
using System.Collections.Generic;

class Solution
{
    static void Main()
    {
        int numberOfElements = int.Parse(Console.ReadLine()); // Number of elements which make up the association table.
        int numberOfFileNames = int.Parse(Console.ReadLine()); // Number Q of file names to be analyzed.

        Dictionary<string, string> file_extensions_association_dict = new Dictionary<string, string>();

        for (int i = 0; i < numberOfElements; i++)
        {
            string[] inputs = Console.ReadLine().Split(' '); // file extension, MIME type.
            file_extensions_association_dict.Add(inputs[0].ToLower(), inputs[1]);
        }

        for (int i = 0; i < nubmerOfFileNames; i++)
        {
            string fileName = Console.ReadLine(); // One file name per line.
            string extension = "";

            if (fileName.IndexOf(".") != -1) // Checking for a dot
                extension = fileName.Substring(fileName.LastIndexOf(".") + 1).ToLower(); // Including a dot

            Console.WriteLine(file_extensions_association_dict.ContainsKey(extension) ? file_extensions_association_dict[extension] : "UNKNOWN");
        }
    }
}

File: Easy/MIME Type/JavaScript/MIME Type.js
numberOfElements = parseInt(readline()); // Number of elements which make up the association table.
numberOfFileNames = parseInt(readline()); // Number Q of file names to be analyzed.

file_extensions_association_array = {};

for (i = 0; i < numberOfElements; i++) {
    inputs = readline().split(' ');
    file_extensions_association_array[(inputs[0].toLowerCase())] = inputs[1];
}

for (i = 0; i < numberOfFileNames; i++) {
    fileName = readline(); // One file name per line.
    extension = "";

    if (fileName.indexOf(".") != -1) // Checking for a dot
        extension = fileName.substring(fileName.lastIndexOf(".") + 1).toLowerCase(); // Including a dot

    print(file_extensions_association_array.hasOwnProperty(extension) ? file_extensions_association_array[extension] : "UNKNOWN");
}